source/test/main.test.ts

```diff
@@ -71,4 +71,22 @@ test("has output - bucketName", () => {
     const stack = new MainStack(app, 'test');
     const template = Template.fromStack(stack);
     template.hasOutput('bucketName', {})
-})
\ No newline at end of file
+})
+
+test("has 1 CustomResource", () => {
+    const app = new App();
+    const stack = new MainStack(app, 'test');
+    const template = Template.fromStack(stack);
+    template.hasResource('AWS::CloudFormation::CustomResource', 1)
```

Comment:
unit test makes sure the CloudFormation resources created with expected properties, such as custom resource handler, the runtime/env/vpc of lambda, etc., see https://github.com/aws/aws-cdk/blob/ea7802b4b5ea4781ed884fc8d330173b192d569d/packages/%40aws-cdk/custom-resources/test/aws-custom-resource/aws-custom-resource.test.ts#L9-L75 for reference